docs/guides/terraform/terratest.mdx

```diff
@@ -0,0 +1,247 @@
+---
+description: Use Terratest to validate the infrastructure of a Nitric Go project deployed with Terraform
+tags:
+  - Terraform
+  - Testing
+---
+
+# Testing AWS resources with Terratest
+
+This guide will walk you through adding [Terratest](https://terratest.gruntwork.io/) to a Nitric project.
+
+## How Terratest works
+
+Terratest is designed to automate the entire process of testing your Terraform code with the following steps:
+
+- **Initialize**: Terratest will automatically run `terraform init` to initialize the Terraform working directory.
+- **Apply**: It will then run `terraform apply` to deploy the infrastructure as defined in your Terraform code.
+- **Assert**: The test script will then run assertions to check that the infrastructure was created as expected.
+- **Teardown**: Finally, it will run `terraform destroy` to tear down the infrastructure after the test completes.
+
+## What we'll be doing
+
+1. Create and set up your application.
+2. Deploying to AWS with a Terraform provider.
+3. Add and execute Terratest.
+
+## Create and set up your application.
+
+Our sample project creates a real-time communication service using [WebSockets](/websockets) and a [key-value store](/keyvalue) for connections.
+
+We intend to deploy to AWS and will use Terratest to ensure that the following:
+
+- **API Gateway WebSocket**: Confirm that the websocket endpoint is correctly configured for real-time communication.
+- **DynamoDB Table**: Verify that the key-value store for connections is created and operational.
+- **IAM Roles**: Ensure that permissions for interacting with AWS services are correctly set up.
+
+Let's start by creating a new project for our application.
+
+```bash
+nitric new my-websocket-app go-starter
+```
+
+### Application code
+
+Replace the contents of `services\hello.go` with our websockets application.
+
+```go
```

Comment:
Maybe replace this with two options:

- follow the existing guide and come back once complete
- clone this github repo with a pre-built nitric project

Reply:
I've added the reference to the existing guide 
The clone option requires a sparse checkout from the examples repo - lets chat about how we can make this easier.